lib/chef/property.rb

```diff
@@ -463,7 +463,12 @@ def coerce(resource, value)
     def validate(resource, value)
       # If we have no default value, `nil` is never coerced or validated
       unless value.nil? && !has_default?
-        (resource || Chef::Mixin::ParamsValidate).validate({ name => value }, { name => validation_options })
+        if resource
+          resource.validate({ name => value }, { name => validation_options })
+        else
+          name = self.name || :property_type
+          (resource || Chef::Mixin::ParamsValidate).validate({ name => value }, { name => validation_options })
```

Comment:
Can we remove the first half of this `||`?  Isn't that captured by the if/else we are in?